waku/node/waku_node.nim

```diff
@@ -917,16 +917,20 @@ proc mountLightPush*(
     pushHandler = proc(
         peer: PeerId, pubsubTopic: string, message: WakuMessage
     ): Future[WakuLightPushResult[void]] {.async.} =
-      let publishedCount =
-        await node.wakuRelay.publish(pubsubTopic, message.encode().buffer)
+      let validationRes = await node.wakuRelay.validateMessage(pubSubTopic, message)
+      if validationRes.isErr():
+        return err(validationRes.error)
+      else:
```

Comment:
Since you return in the previous line, I don't think this `else` is necessary. In other words, if validation failed, this block of code will not be executed, whether in an `else` block or not.